cliutil/log.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 The AthanorLabs/atomic-swap Authors
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package cliutil
+
+import (
+	"fmt"
+
+	logging "github.com/ipfs/go-log"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	// FlagLogLevel is the log level flag.
+	FlagLogLevel = "log-level"
+)
+
+// SetLogLevelsFromContext sets the log levels for all packages from the CLI context.
+func SetLogLevelsFromContext(c *cli.Context) error {
+	const (
+		levelError = "error"
+		levelWarn  = "warn"
+		levelInfo  = "info"
+		levelDebug = "debug"
+	)
+
+	level := c.String(FlagLogLevel)
+	switch level {
+	case levelError, levelWarn, levelInfo, levelDebug:
+	default:
+		return fmt.Errorf("invalid log level %q", level)
+	}
+
+	setLogLevels(level)
+	return nil
+}
+
+func setLogLevels(level string) {
```

Comment:
Maybe now would be an ideal time to move this function to somewhere like `common` so we don't have to synchronize the entries in so many different places?

Reply:
where else is this used? i thought it was just swapd and now bootnode

Reply:
4 places: Here, `swapd`, `swaptester` and `daemon/swap_daemon_test.go`.